syntax = "proto3";

service FileServer {
  rpc upload_batch(BatchChunk) returns (Reply) {}
}

message BatchChunk {
    repeated Chunk chunks = 1;
}

message Chunk {
  bytes buffer = 1;
  string name = 2;
  int32 identification_number = 3;
}

message Request {
  string name = 1;
}

message Reply {
  string result = 1;
}